core/shared/src/main/scala/scalaz/zio/internal/FiberContext.scala

```diff
@@ -312,6 +321,16 @@ private[zio] final class FiberContext[E, A](
   private[this] final def getDescriptor: Fiber.Descriptor =
     Fiber.Descriptor(fiberId, state.get.interrupted, executor)
 
+  // We make a copy of the supervised fibers set as an array
+  // to prevent mutations of the set from propagating to the caller.
+  private[this] final def getFibers: Iterable[Fiber[_, _]] =
```

Comment:
Ah, it's `Array` not `List`. That will be faster, of course, and does restrict us to `Iterable`, or at best `IndexedSeq`.

Reply:
Good point- refined the `Iterable` to `IndexedSeq`.